### PR TITLE
Remove section share auth

### PR DESCRIPTION
**Pull request by grumbach**
Removes `AuthKind::SectionShare` and all the awkwardness that came with it.

### CODE REVIEW

File: sn_node/src/comm/listener.rs [hunk 1]
```diff
@@ -67,8 +67,6 @@ impl MsgListener {
                     };
```
Comment (grumbach) [inline sn_node/src/comm/listener.rs:68]: This is now solved!

File: sn_node/src/comm/listener.rs [hunk 2]
```diff
@@ -67,8 +67,6 @@ impl MsgListener {
...
 
                     let src_name = match wire_msg.auth() {
-                        // NB TODO is default acceptable??
-                        AuthKind::SectionShare(_) => Default::default(),
                         AuthKind::Client(auth) => auth.public_key.into(),
                         AuthKind::Node(auth) => {
```
Comment (grumbach) [inline sn_node/src/comm/listener.rs:71]: Now I'm happy again 🕺

**Review by davidrusu — APPROVED** [review]
:fire:

**Pull request merged**